crates/bevy_window/src/system.rs

```diff
@@ -25,33 +55,31 @@ pub fn exit_on_all_closed(mut app_exit_events: EventWriter<AppExit>, windows: Re
 ///
 /// [`WindowPlugin`]: crate::WindowPlugin
 pub fn close_when_requested(
-    mut windows: ResMut<Windows>,
     mut closed: EventReader<WindowCloseRequested>,
+    mut window_closed: EventWriter<WindowClosed>,
 ) {
     for event in closed.iter() {
-        windows.get_mut(event.id).map(Window::close);
+        window_closed.send(WindowClosed {
+            entity: event.entity,
+        });
```

Comment:
Right now the `WindowClosed`-event is being used as "Please close this window" and not "window _has_ been closed"

Either the `WindowClosed` event should be renamed to "WindowCloseRequest", but we already have that.
or we should not be emitting this event here?